django_orghierarchy/migrations/0001_initial.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.7 on 2017-11-13 14:02
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.db.models.deletion
+import swapper
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        swapper.dependency('django_orghierarchy', 'DataSource')
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='DataSource',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
```

Comment:
How does swapper behave if the project we are plugging this into (in this case, linkedevents) has a different id field?

Or is it so that if such a model is plugged in, then this migration will not be run at all so we don't have to care about the specifics here?

Reply:
Haven't test it out, I think I will add some integration tests later. But for linkedevents project, I think we can just plugin the model without inheriting from the `AbstractDataSource`.

The data source model in `linkedevents` is a bit special in the sense that it uses a `CharField` as primary key. I think it may not be a good idea to borrow the idea to a generic app, so the default primary key is used here.